common/transport/mqtt/src/main/java/org/thingsboard/server/transport/mqtt/MqttTransportHandler.java

```diff
@@ -953,6 +998,30 @@ private void checkGatewaySession(SessionMetaData sessionMetaData) {
         }
     }
 
+    private void checkSparkPlugConnected(SessionMetaData sessionMetaData, MqttConnectMessage connectMessage) {
+        if (((MqttDeviceProfileTransportConfiguration) deviceSessionCtx.getDeviceProfile().getProfileData().getTransportConfiguration())
+                .isSparkPlug()) {
+            TransportDeviceInfo device = deviceSessionCtx.getDeviceInfo();
+            try {
+                JsonNode infoNode = context.getMapper().readTree(device.getAdditionalInfo());
+                if (infoNode != null) {
+                    SparkplugTopic sparkplugTopic = parseTopic(connectMessage.payload().willTopic());
+                    if (sparkPlugSessionHandler == null) {
+                        log.error("Connected [{}] [{}]", sparkplugTopic.isNode() ? "node" : "device", sparkplugTopic.getType());
+                        sparkPlugSessionHandler = new SparkplugNodeSessionHandler(deviceSessionCtx, sessionId, sparkplugTopic.toString());
+                    } else {
+                        log.error("ReConnected [{}] [{}]", sparkplugTopic.isNode() ? "node" : "device", sparkplugTopic.getType());
+                     }
+                    if (infoNode.has(DefaultTransportService.OVERWRITE_ACTIVITY_TIME) && infoNode.get(DefaultTransportService.OVERWRITE_ACTIVITY_TIME).isBoolean()) {
```

Comment:
Is this a copy-paste code from the gateway method? What does this OVERWRITE_ACTIVITY_TIME means? Please remove this if it is copy-paste. Looks like you don't need to parse the 
`JsonNode infoNode = context.getMapper().readTree(device.getAdditionalInfo());` here.

Reply:
Ok